src/cnaas_nms/api/generic.py

```diff
@@ -56,6 +58,51 @@ def offset_results() -> int:
     return offset
 
 
+def pagination_headers(total_count) -> dict:
+    per_page = DEFAULT_PER_PAGE
+    page_arg = 1
+    links = []
+    headers = {
+        "X-Total-Count": total_count,
+    }
+
+    args = request.args
+    if 'per_page' in args:
+        try:
+            per_page_arg = int(args['per_page'])
+            assert 1 <= per_page_arg <= MAX_PER_PAGE
+            per_page = per_page_arg
+        except (AssertionError, ValueError):
+            pass
+
+    last_page = int(math.ceil(total_count / per_page))
+    if last_page == 1:
+        return {}
```

Comment:
Don't we still want the X-Total-Count header?

Reply:
Yes indeed! Good catch!